crates/eww/src/app.rs

```diff
@@ -200,7 +203,20 @@ impl App {
     }
 
     fn update_state(&mut self, fieldname: VarName, value: DynVal) {
-        self.eww_state.update_variable(fieldname, value)
+        self.eww_state.update_variable(fieldname.clone(), value);
+
+        if let Some(linked_poll_vars) = self.eww_config.poll_var_trigger.get(&fieldname) {
+            linked_poll_vars.iter().filter_map(|name| self.eww_config.get_script_var(name).ok()).for_each(|var| {
+                if let ScriptVarDefinition::Poll(poll_var) = var {
+                    match poll_var.trigger_expr.eval(self.eww_state.get_variables()).map(|v| v.as_bool()) {
+                        Ok(Ok(true)) => self.script_var_handler.add(var.clone()),
+                        Ok(Ok(false)) => self.script_var_handler.stop_for_variable(poll_var.name.clone()),
+                        Ok(Err(err)) => error_handling_ctx::print_error(anyhow!(err)),
+                        Err(err) => error_handling_ctx::print_error(anyhow!(err)),
+                    };
+                }
+            });
+        }
```

Comment:
this is somewhat ugly - It should be easily possible to flatten the structure here by using and_then to not have the nested results. Also, there is not all that much reason to go with for_each over a simple for-loop here

Reply:
I don't think it'll remove any ugliness, `match` expr does this with 2 lines and 2 line for two different errors. Using `map_err` will only map the inner error, and outer error still need to be mapped using `anyhow!`. Also since line is long, if-else in the `and_then` will take 4-5 lines anyway 🤔🤔.

Reply:
`and_then` won't accept this as `ConversionError` is different than `EvalError` and incompatible, it'll involve anyhow inside and once outside, increasing complexity of reading the code itself. I think its pretty explicit and readable right now 🤔🤔

Reply:
true, in that case this is good enough for now 👍🏻